pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -464,9 +464,13 @@ func (s *DiscoveryServer) WorkloadUpdate(shard, id string, workloadLabels map[st
 	adsClientsMutex.RLock()
 	for _, connection := range adsClients {
 		// update node label
-		if connection.modelNode.IPAddresses[0] == id {
-			// set to nil, trigger re-set in SetWorkloadLabels
-			connection.modelNode.WorkloadLabels = nil
+		if connection.modelNode.IPAddresses[0] == ip {
+			select {
+			// trigger re-set in SetWorkloadLabels
+			case connection.updateChannel <- &UpdateEvent{workloadLabel: true}:
```

Comment:
why not block here? we may miss updated this way

Reply:
the channel is buffered with capacity 1, if this blocks, there is already labels updates un-processed.

Reply:
hm I guess so. if we extend this updateChannel with future update types we need to be careful though, but I think it's fine for now